Dockerfile

```
@@ -18,15 +18,32 @@ COPY --chown=app . .
 RUN npm config set update-notifier false
 RUN npm ci --ignore-scripts && npm run compile
 
-RUN if [ "$SKIP_TESTS" = 'false' ]; then ln -s $APP_HOME/tests/fixtures env && npm test && rm env; fi
+RUN set -e && \
+    npm config set update-notifier false && \
+    npm ci --ignore-scripts && \
+    npm run compile
+
+# Run tests with the CI-specific script that has proper Jest flags
+RUN set -e && \
+    if [ "$SKIP_TESTS" = 'false' ]; then \
+        echo "Setting up test environment..." && \
+        ln -s $APP_HOME/tests/fixtures env && \
+        echo "Running CI tests..." && \
+        npm run test:ci && \
+        echo "Cleaning up test environment..." && \
+        rm env && \
+        echo "Tests completed successfully"; \
```

Could this be simplified? I am not sure if the symlink is necessary. Maybe
```suggestion
        echo "Running CI tests..." && \
        ENV_DIR=$APP_HOME/tests/fixtures npm run test:ci && \
        echo "Tests completed successfully"; \
```
would do.

---

I'll remove the symlink and set the env at line 11.